app/src/pages/[resourceType]/[id].tsx

```diff
@@ -80,6 +87,32 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     }
   );
 
+  //Sorts the dependencies of an artifact alphabetically based on display name. If a dependency has a link, however,
+  //it will appear at the top of the list.
+  useEffect(() => {
+    dataRequirements?.relatedArtifact?.sort((a, b) => {
+      const displayA = a?.display?.toUpperCase();
+      const displayB = b?.display?.toUpperCase();
+      if (displayA !== undefined && displayB !== undefined) {
+        if (
+          a.resource?.includes('Library') ||
+          a.resource?.includes('Measure') ||
+          b.resource?.includes('Library') ||
+          b.resource?.includes('Measure')
+        ) {
+          return 1;
+        }
+        if (displayA < displayB) {
+          return -1;
+        }
+        if (displayA > displayB) {
+          return 1;
+        }
+      }
+      return 0;
+    });
+  }, [dataRequirements?.relatedArtifact]);
```

Comment:
This works perfectly fine, but a slight nitpick is that I think it might be more intuitive if we don't sort the relatedArtifact within a useEffect whose dependency is itself the relatedArtifact. It could be a bit confusing if one is assuming that the dependencies are rendered as is and then having to trace down where the side effect of the `.sort` call actually happens on that original array.

My suggestion is to make use of the fact that `.sort` in javascript also [returns a reference to the sorted array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#return_value), and be more expressive about the variable names. So after the `dataRequirements` variable is initialiazed after the trpc useQuery is done, you could do something like:

```ts
  const sortedDependencies = dataRequirements?.relatedArtifact?.sort((a, b) => {
      // Your original code
  }) ?? []; // default to empty array until we have actual data to sort through
```

Then, when you go to render them, instead of

```tsx
                {dataRequirements.relatedArtifact.map((relatedArtifact, index) => (
                  <Dependency key={index} relatedArtifact={relatedArtifact} />
                ))}
```

You can now do

```tsx
                {sortedDependencies.map((relatedArtifact, index) => (
                  <Dependency key={index} relatedArtifact={relatedArtifact} />
                ))}
```

Now looking at the code, it is clear that the dependencies are expected to be iterated on in a sorted order, and I can easily take a look at the variable declaration of `sortedDependencies` to see how that is done.